code/modules/projectiles/guns/energy/crank.dm

```diff
@@ -0,0 +1,43 @@
+/obj/item/weapon/gun/energy/rifle/icelance
+	name = "icelance rifle"
+	desc = "A Tajaran made rifle, it houses a crank-chargable internal battery. It only holds three shots and each shot must be cranked manually."
+	icon_state = "icelance"
+	item_state = "icelance"
+	fire_sound = 'sound/weapons/Laser.ogg'
+	max_shots = 3
+	fire_delay = 10
+	fire_delay_wielded = 8
+	origin_tech = list(TECH_COMBAT = 1, TECH_MAGNET = 1)
+	projectile_type = /obj/item/projectile/beam
+	secondary_projectile_type = null
+	secondary_fire_sound = null
+	var/is_charging = FALSE
+
+	firemodes = list()
+	modifystate = null
+
+/obj/item/weapon/gun/energy/rifle/icelance/attack_self(mob/living/user as mob)
+	if(is_charging)
+		user << "<span class='warning'>You are already charging \the [src].</span>"
+		return
+	if(power_supply.charge < power_supply.maxcharge)
+		user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+		user.visible_message(
+				"<span class='warning'>\The [user] begins to rotate \the [src]'s crank!</span>",
```

Comment:
While this is technically valid, this grammar seems a bit clunky. 

This message might also be better suited to notice instead of warning.